lib/apps.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+const ApiGroup = require('./api-group');
+
+class Apps extends ApiGroup {
+  constructor(options) {
+    const genericTypes = [
+      // Deprecated name of statefulsets in kubernetes 1.4
+      'petsets',
```

Comment:
I almost feel it might be worthwhile to have a `deprecatedTypes` array that we can pass to the constructor. Something like:
```
const deprecatedTypes = [ { name: 'petsets', endorsed: 'statefulsets' } ]

options = Object.assign({}, options {
  ...
  deprecatedTypes: deprecatedTypes
});
```

Then in ApiGroup we could have something like:
```
for (const type of options.deprecatedTypes) {
  this[type] = new BaseObject({
    name: type.deprecated,
    parentPath: this.path,
    api: this,
   message: `${ type.deprecated } is a deprecated name. Please use ${ type.endorsed }`
  });
}
```

Then somewhere in the base class we could check for the `message` property and do a `console.warn` with the value in `message` if it exists. Probably in the `CallableObject` class.

Let me know what you think.